src/textprop.c

```diff
@@ -497,7 +497,16 @@ prop_add_common(
 	end_col = 1;
 
     if (dict_has_key(dict, "id"))
-	id = dict_get_number(dict, "id");
+    {
+	vimlong_T x;
+	x = dict_get_number(dict, "id");
+	if (x > INT_MAX)
+	{
+	    semsg(_(e_val_too_large), dict_get_string(dict, "id", FALSE));
+	    goto theend;
+	}
+	id = (int)x;
```

Comment:
Related to adding text properties, I see a missing closing parenthesis in an example in the doc:
```
diff --git a/runtime/doc/textprop.txt b/runtime/doc/textprop.txt
index 6b46e06df..012d7686b 100644
--- a/runtime/doc/textprop.txt
+++ b/runtime/doc/textprop.txt
@@ -271,7 +271,7 @@ prop_add_list({props}, [{item}, ...])                       *prop_add_list()*
                        call prop_add_list(#{type: 'MyProp', id: 2},
                                        \ [[1, 4, 1, 7],
                                        \  [1, 15, 1, 20],
-                                       \  [2, 30, 3, 30]]
+                                       \  [2, 30, 3, 30]])
 <
                Can also be used as a |method|: >
                        GetProp()->prop_add_list([[1, 1, 1, 2], [1, 4, 1, 8]])
```

Reply:
I just see that `prop_add_list()` also crashes after this patch. E.g.

```
:help 
:call prop_type_add('my_test', {})
:call prop_add_list(#{type: 'my_test', id: 2147483648}, [[1, 1, 1, 1]])

Vim: Caught deadly signal SEGV
Vim: Finished.
Segmentation fault: 11
```
It requires the same kind of fix at textprop.c:375 in `f_prop_add_list()`:
```
   374     if (dict_has_key(dict, "id"))
   375         id = dict_get_number(dict, "id");
```

Reply:
thanks, fixed both as well. Also tested for negative INT_MIN